contrib/mesos/pkg/queue/delay/policy.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package delay
+
+// Decide whether an item in a delay-queue should be rescheduled if an item with the
+// same UniqueID is added/offered. Rescheduling does not effect the Item value.
+type SchedulingPolicy int
+
+const (
+	PreferLatest SchedulingPolicy = iota
+	PreferEarliest
+)
+
+func (dp SchedulingPolicy) EventTime(a, b Priority) (result Priority) {
```

Comment:
would a better name for this func be `Prioritize`? `EventTime` is not very informative as to what this is actually doing